### PR TITLE
refactor(sbb-skiplink-list): lit migration

### PR DESCRIPTION
**Pull request by federicoisepponfincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- migrate `sbb-skiplink` to lit

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-skiplink-list/sbb-skiplink-list.tsx [hunk 1]
```diff
@@ -78,31 +79,39 @@ export class SbbSkiplinkList implements ComponentInterface {
...
+          ${this._links.map(
+            (_, index) =>
+              html` <li>
+                <slot name=${`link-${index}`} @slotchange=${(): void => this._readLinks()} />
+              </li>`,
+          )}
         </ul>
         <span hidden>
-          <slot onSlotchange={() => this._readLinks()} />
+          <slot @slotchange=${() => this._readLinks()} />
```
Comment (TomMenga) [inline src/components/sbb-skiplink-list/sbb-skiplink-list.tsx:105]:
```suggestion
          <slot @slotchange=${() => this._readLinks()}></slot>
```

**Review by TomMenga — CHANGES_REQUESTED** [review]
Good job,
A couple of fixes and a file to revert and you're good to go

**Comment by TomMenga** [comment]
Another necessary step is to rebase from lit-migration and to run `yarn integrity`

File: src/components/sbb-breadcrumb-group/sbb-breadcrumb-group.e2e.ts
[the remark this answers is not in the document]
Comment (federicoisepponfincons) [reply src/components/sbb-breadcrumb-group/sbb-breadcrumb-group.e2e.ts:1]: fixed

File: src/components/sbb-skiplink-list/sbb-skiplink-list.e2e.ts [hunk 2]
```diff
@@ -1,66 +1,62 @@
-import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
+import { assert, expect, fixture } from '@open-wc/testing';
+import { html } from 'lit/static-html.js';
+import { waitForLitRender } from '../../global/testing';
+import { SbbSkiplinkList } from './sbb-skiplink-list';
+import { SbbLink } from '../sbb-link';
+import '../sbb-link';
+import './sbb-skiplink-list';
```
Comment (jeripeierSBB) [inline src/components/sbb-skiplink-list/sbb-skiplink-list.e2e.ts:7]: this import is not needed 
```suggestion
```

File: src/components/sbb-skiplink-list/sbb-skiplink-list.e2e.ts [hunk 3]
```diff
@@ -1,66 +1,62 @@
...
   });
 
   it('should detected later added links', async () => {
-    page = await newE2EPage();
-    await page.setContent(`<sbb-skiplink-list></sbb-skiplink-list>`);
-    element = await page.find('sbb-skiplink-list');
+    element = await fixture(html`<sbb-skiplink-list></sbb-skiplink-list>`);
 
-    await page.waitForChanges();
+    await waitForLitRender(element);
```
Comment (jeripeierSBB) [inline src/components/sbb-skiplink-list/sbb-skiplink-list.e2e.ts:51]: should not be needed
```suggestion
```

File: src/components/sbb-skiplink-list/sbb-skiplink-list.tsx [hunk 4]
```diff
@@ -1,51 +1,50 @@
...
 } from '../../global/eventing';
+import { CSSResult, html, LitElement, nothing, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { spread } from '@open-wc/lit-helpers';
+import { SbbLink } from '../sbb-link';
+import Style from './sbb-skiplink-list.scss?lit&inline';
+import '../sbb-title';
 
 /**
  * @slot unnamed - Use this to provide links for the list.
```
Comment (jeripeierSBB) [inline src/components/sbb-skiplink-list/sbb-skiplink-list.tsx:15]: new notation. but you anyway need to rebase to lit-migration branch, there is already an updated string
```suggestion
 * @slot - Use this to provide links for the list.
```

**Review by jeripeierSBB — COMMENTED** [review]
Some minor comments :)

File: src/components/sbb-skiplink-list/sbb-skiplink-list.tsx (hunk 4 above)
Comment (federicoisepponfincons) [reply src/components/sbb-skiplink-list/sbb-skiplink-list.tsx]: already corrected in the lit-migration branch :)

File: src/components/sbb-timetable-row-column-headers/readme.md [hunk 5]
```diff
@@ -18,3 +18,27 @@ here to show the various configuration options to component developers. !!
 
 <hr/>
 
+# Variables
```
Comment (jeripeierSBB) [inline src/components/sbb-timetable-row-column-headers/readme.md:21]: strange changes?

File: src/components/sbb-skiplink-list/sbb-skiplink-list.tsx [hunk 6]
```diff
@@ -1,51 +1,50 @@
...
 
   /** State of listed named slots, by indicating whether any element for a named slot is defined. */
-  @State() private _namedSlots = createNamedSlotState('title');
-
-  @Element() private _element!: HTMLElement;
+  @state() private _namedSlots = createNamedSlotState('title');
 
-  private _syncLinks(): void {
-    this._element.querySelectorAll('sbb-link').forEach((link: HTMLSbbLinkElement) => {
+  public syncLinks(): void {
```
Comment (jeripeierSBB) [inline src/components/sbb-skiplink-list/sbb-skiplink-list.tsx:33]: This method probably should be private. or is there any reason to have it public?

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

**Pull request merged**